atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/Schema.java

```diff
@@ -109,6 +112,19 @@ public void addTableDefinition(String tableName, TableDefinition definition) {
         Preconditions.checkArgument(
                 Schemas.isTableNameValid(tableName),
                 "Invalid table name " + tableName);
+        if (!ignoreTableNameLengthChecks) {
+            int internalTableNameLength = Schemas.getInternalTableNameLength(tableName, namespace);
+            Preconditions.checkArgument(
+                    internalTableNameLength < POSTGRES_TABLE_NAME_CHAR_LIMIT,
+                    "Table name %s is too long, exceeds Cassandra and Postgres limits. " +
+                            "If running using a different KVS, set the ignoreTableNameLength flag.",
```

Comment:
Actually I think this is correct - if we flip them then one bad for both C* and Postgres will print the C* message.

Ideally we'd avoid ordering here though I'm not sure it's worth the complexity that might add (`"Table name %s is too long, known to exceed limits for the following KVSs: %s"`?)